### PR TITLE
[FIX] Separate dev dependencies from run-time dependencies

### PR DESCRIPTION
**Pull request by minasmart**
This also fixes a couple bugs in running this thing as a node module.

Additionally, the package gets built on pre-publish

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,12 @@
   "author": "Shopify Inc.",
   "license": "MIT",
   "dependencies": {
+    "node-fetch": "1.3.3"
+  },
+  "devDependencies": {
```
Comment (richgilbank) [inline package.json]: 😻

File: src/metal/global.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+let globalNamespace;
+
+if (typeof global === 'undefined') {
+  globalNamespace = window;
+} else {
+  globalNamespace = global;
+}
+
+export default globalNamespace;
```
Comment (richgilbank) [inline src/metal/global.js]: Could this be simplified to `export default global || window`?

**Comment by richgilbank** [comment]
@tessalt @lreeves Can one of you two take a peek at this as well?

File: src/metal/global.js (hunk 2 above)
Comment (minasmart) [inline src/metal/global.js]: Nah. If global is undefined that will throw. Also, no logic in an export statement.

**Comment by lreeves** [comment]
It looks good as far as I can tell!

**Pull request merged**